### ISSUE TITLE
nprocesses>1 (again): should cudacpp implement this or not? (if needed, implement it)

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#940]
Hi @oliviermattelaer this is a followup to #872 and PR #935 (and another ~10 issues and PR previous to that).

I file this as a separate ticket to hopefully make it clearer.

(1) Question assigned to you @oliviermattelaer, because only you have the answer: do we need to implement nprocesses>1 in cudacpp or not? Reminder: cudacpp assumes nprocesses==1, but then we also need fortran to have mirrorprocs=false. Can you confirm that we can keep nprocesses=1 in cudacpp and mirrorprocs=false in fortran (i.e. undo #754)? If yes, can you confirm that this is because we generate separtely the P directories for mirro processes (e.g. pp_tt012j has BOTH P2_gu_ttxgu AND P2_gux_ttxgux as SEPARATE directories).

(2) If instead the answer is no, and we do need to implement nprocesses>1 (i.e. add arrays and IMIRROR=1,2 or 0,1 to cudacpp), I assign to me for implementing this.

Thanks
 Andrea

**Comment by valassi** [comment]
See the discussion in #872 and PR #935 (fixing the number of reset_cumuative_variable call - also related to #486 and https://github.com/oliviermattelaer/mg5amc_test/issues/22) 

Especially see https://github.com/madgraph5/madgraph4gpu/issues/872#issuecomment-2251160179: MIRRORPROCS=true is needed to get the right cross sections, and this comes with nprocesses=2, but there is no need to add arrays. 

The ONLY place where I see an array(nprocesses=2) could be needed is the recomputation of helicities, if we wanted to have one per mirror, as discussed in #872, see Olivier's comments. The patch in #935 however assumes that we do not need that, i.e. that the two helicity lists are identical, and it computes helicities only once. So we do not need to have two lists of helicities for the two mirrors.

I keep this open, maybe the only thing that can be done is
- add more comments around the static assert for nprocesses=2, pointing here and #872 
- add another sanity check that imirror can be only 1 or 2, an dthat if it is 2 it is with nprocesses=2 (rephrase, assert imirror <= nprocesses)

**Comment by valassi** [comment]
I guess that, as per Olivier's conclusions in #941, there is no need to do anything with nprocesses>1. Eventually the second helicity calculationf ror the two mirrors will be removed also in fortran IIUC.

Keep this open for th emoment until that issue is completed
